src/wp-includes/formatting.php

```diff
@@ -960,19 +960,7 @@ function _wp_specialchars( $text, $quote_style = ENT_NOQUOTES, $charset = false,
 		$quote_style = ENT_QUOTES;
 	}
 
-	// Store the site charset as a static to avoid multiple calls to wp_load_alloptions().
-	if ( ! $charset ) {
-		static $_charset = null;
-		if ( ! isset( $_charset ) ) {
-			$alloptions = wp_load_alloptions();
-			$_charset   = isset( $alloptions['blog_charset'] ) ? $alloptions['blog_charset'] : '';
-		}
-		$charset = $_charset;
-	}
-
-	if ( in_array( $charset, array( 'utf8', 'utf-8', 'UTF8' ), true ) ) {
-		$charset = 'UTF-8';
-	}
+	$charset = _canonical_charset( $charset || get_option( 'blog_charset' ) );
```

Comment:
There's an issue here. On a test site, before this change, this function starts with a `$charset=false` value, then processes options and has `$charset = "UTF-8"` value. After this change, the charset starts as `false` and becomes `true` after this line.

> Warning: htmlspecialchars(): Charset "1" is not supported, assuming UTF-8 in /var/www/html/wp-includes/formatting.php on line 984

Reply:
Took me a minute to figure this out, but I've made this mistake many times before. The `||` expression returns a boolean value, not the first non-`false` value. I've updated in b9347d7f2d to use a proper ternary.

Reply:
That fixed it for me 🙂